### PR TITLE
feat(escalating-issues): Wrap Group updates in db transaction

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
We are getting timeouts when executing the update SQL query. This PR wraps that method in a database transaction. This ensures atomicity and can improve performance for grouped write operations.

### CODE REVIEW

File: src/sentry/models/group.py [hunk 1]
```diff
@@ -421,26 +421,30 @@ def update_group_status(
...
+                .select_for_update()
+            )
 
-        for group in to_be_updated:
-            group.status = status
-            group.substatus = substatus
+            for group in selected_groups:
+                group.status = status
+                group.substatus = substatus
+                groups_list.append(group)
```
Comment (lobsterkatie) [inline src/sentry/models/group.py:435]: This is probably just my lack of knowledge about the django ORM, but why do the groups need to be moved from `selected_groups` into a separate list? Is that so the queryset gets materialized? Would it be equivalent to just wrap the `Group.objects...` expression in a list? (Not saying that's necessarily better, just trying to understand what's happening.)

File: src/sentry/tasks/auto_ongoing_issues.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 logger = logging.getLogger(__name__)
 
 TRANSITION_AFTER_DAYS = 7
-ITERATOR_CHUNK = 10_000
+ITERATOR_CHUNK = 500
```
[the remark this answers is not in the document]
Comment (armenzg) [reply src/sentry/tasks/auto_ongoing_issues.py:25]: I encourage you to separate this change into its own PR and deploy them separately so you can determine what was the correct fix.

**Review by armenzg — APPROVED** [review]
Neat to learn about [select for update](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#select-for-update)!

File: src/sentry/models/group.py (hunk 1 above)
Comment (lobsterkatie) [reply src/sentry/models/group.py:435]: Got it, thanks. Perhaps name it `modified_groups`, then, though?

**Review by lobsterkatie — APPROVED** [review]
LGTM!

File: src/sentry/models/group.py [hunk 3]
```diff
@@ -421,26 +421,30 @@ def update_group_status(
...
 
-        to_be_updated = self.filter(id__in=[g.id for g in groups]).exclude(
-            status=status, substatus=substatus
-        )
+        groups_list = []
+        with transaction.atomic(router.db_for_write(Group)):
+            selected_groups = (
+                Group.objects.filter(id__in=[g.id for g in groups])
+                .exclude(status=status, substatus=substatus)
+                .select_for_update()
```
[the remark this answers is not in the document]
Comment (wedamija) [reply src/sentry/models/group.py:429]: I'm not sure it's a great idea to lock rows in the `Group` table. Would it make sense to add a lock around this function with the org id as the key?

Is the problem you're trying to solve duplicate activity rows?

**Pull request merged**